midend/predication.cpp

```diff
@@ -13,11 +13,158 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+#include <algorithm>
 
 #include "predication.h"
 #include "frontends/p4/cloner.h"
 
+
 namespace P4 {
+/// convert an expression into a string that uniqely identifies the value referenced
+/// return null cstring if not a reference to a constant thing.
+static cstring expr_name(const IR::Expression *exp) {
+    if (auto p = exp->to<IR::PathExpression>())
+        return p->path->name;
+    if (auto m = exp->to<IR::Member>()) {
+        if (auto base = expr_name(m->expr))
+            return base + "." + m->member;
+    } else if (auto a = exp->to<IR::ArrayIndex>()) {
+        if (auto k = a->right->to<IR::Constant>())
+            if (auto base = expr_name(a->left))
+                return base + "." + std::to_string(k->asInt()); }
+    return cstring();
+}
+
+bool Predication::ReplaceChecker::preorder(const IR::AssignmentStatement * statement) {
+    if (!statement->right->is<IR::Mux>()) {
+        conflictDetected = true;
+        return false;
+    }
+    assignmentStatement = statement;
+    visit(statement->right);
+    return false;
+}
+
+bool Predication::ReplaceChecker::preorder(const IR::Mux * mux) {
+    ++currentNestingLevel;
+    auto leftAssignmentName = expr_name(assignmentStatement->left);
```

Comment:
_ReplaceChecker_ was used for checking if the expression(right side of the assignment) could be placed in nested ternary operators. This Visitor is unnecessary because we print all dependencies of the Assignment Statement so I removed it.